components/eam/src/physics/rrtmgp/cam_optics.F90

```diff
@@ -503,9 +507,15 @@ subroutine set_aerosol_optics_sw(icall, state, pbuf, &
       tau_w = 0._r8
       tau_w_g = 0._r8
       tau_w_f = 0._r8
-      call aer_rad_props_sw(icall, state, pbuf, &
-                            count(night_indices > 0), night_indices, is_cmip6_volc, &
-                            tau, tau_w, tau_w_g, tau_w_f)
+      if(present(clear_rh)) then
```

Comment:
I believe these logical blocks are unnecessary, you should just be able to pass clear_rh and avoid the duplicated code.